sync/src/relayer/transaction_hashes_process.rs

```diff
@@ -0,0 +1,111 @@
+use crate::relayer::compact_block::TransactionHashes;
+use crate::relayer::Relayer;
+use ckb_core::transaction::ProposalShortId;
+use ckb_logger::{debug_target, trace_target};
+use ckb_network::PeerIndex;
+use ckb_protocol::RelayTransactionHashes as FbsRelayTransactionHashes;
+use ckb_store::ChainStore;
+use failure::Error as FailureError;
+use numext_fixed_hash::H256;
+use std::convert::TryInto;
+
+pub struct TransactionHashesProcess<'a, CS> {
+    message: &'a FbsRelayTransactionHashes<'a>,
+    relayer: &'a Relayer<CS>,
+    peer: PeerIndex,
+}
+
+impl<'a, CS: ChainStore + 'static> TransactionHashesProcess<'a, CS> {
+    pub fn new(
+        message: &'a FbsRelayTransactionHashes,
+        relayer: &'a Relayer<CS>,
+        peer: PeerIndex,
+    ) -> Self {
+        TransactionHashesProcess {
+            message,
+            relayer,
+            peer,
+        }
+    }
+
+    pub fn execute(self) -> Result<(), FailureError> {
+        let transaction_hashes: TransactionHashes = (*self.message).try_into()?;
+
+        let mut transit_hashes: Vec<H256> = {
+            let tx_filter = self.relayer.shared().tx_filter();
+            transaction_hashes
+                .hashes
+                .into_iter()
+                .filter(|tx_hash| {
+                    if tx_filter.contains(&tx_hash) {
+                        debug_target!(
```

Comment:
Most of the `transaction_hash` is duplicated. So I would suggest removing this logging.

Reply:
I am not sure, just keep the before logic.

Reply:
would remove it later.